<launch>

<param name="/use_sim_time" value="True"/>
<arg name="path" default="/home/parallels/" />
<arg name="file_1" default="Odom_C" />
<arg name="file_2" default="Acc_C" />
<arg name="file_3" default="Gyro_C" />
<arg name="file_4" default="Mag_C" />
<node pkg="rosbag" type="play" name="player" args="--clock   $(arg path)$(arg file_1).bag $(arg path)$(arg file_2).bag $(arg path)$(arg file_3).bag $(arg path)$(arg file_4).bag  " />

<node name="concat_imu" pkg="concat_imu"  type="concat_imu.py" output="screen"/>

<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args=" 0 0 0 0 0 0 base_footprint imu 400" />


<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
  <param name="output_frame" value="odom_combined"/>
  <param name="base_footprint_frame" value="base_footprint"/>
  <param name="freq" value="60.0"/>
  <param name="odom_used" value="false"/>
  <param name="imu_used" value="true"/>
  <param name="vo_used" value="true"/>
  <param name="debug" value="true"/>
  <remap from="vo" to="/odom" />
  <remap from="imu_data" to="imu/data" />

</node>

<node name="path_node" pkg="path_node"  type="path_node.py" output="screen"/>


<node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node" output="screen" args="_use_magnetic_fld_msg:=true _publish_debug_topics:=true _publish_tf:=false stateless:=false" />

</launch>
